File: AIAPI.py
import requests
import json

# Set the API URL
url = "https://llm.kindo.ai/v1/chat/completions"

# Set your API key and model
api_key = "22c4e203-2032-43c9-9b0e-dfc88270b924-2e7e85682c16884d"  # Replace with your actual KINDO API key
model = "claude-3-5-sonnet-20240620"  # Replace with the model name

# Prepare the data to send in the POST request
data = {
    'model': model,
    'messages': [
        {
            'role': 'user',
            'content': 'hello! please just say hello back. thats it'
        }
    ]
}

# Set the headers
headers = {
    'content-type': 'application/json',
    'api-key': api_key
}

# Make the POST request
response = requests.post(url, json=data, headers=headers)

# Check for errors
if response.status_code == 200:
    response_data = response.json()
    
    # Check if the response contains the expected data
    if 'choices' in response_data and isinstance(response_data['choices'], list) and response_data['choices']:
        ai_response = response_data['choices'][0]['message']['content']
        print('AI Response:', ai_response)
    else:
        print('Unexpected response format or empty response.')
else:
    print('Error:', response.status_code, response.text)

File: aiAPI.php
<?php
$ch = curl_init();


// Set the API URL
$url = "https://llm.kindo.ai/v1/chat/completions";

// Set your API key and model
$apiKey = "22c4e203-2032-43c9-9b0e-dfc88270b924-2e7e85682c16884d";  // Replace with your actual KINDO API key
$model = "claude-3-5-sonnet-20240620";      // Replace with the model name


// Prepare the data to send in the POST request
$data = array(
    'model' => $model,
    'messages' => array(
        array(
            'role' => 'user',
            // 'content' => 'does the microsoft graph api for outlook emails allow developers to access email content from the api?'
            'content' => 'hello! please just say hello back. thats it'
        )
    )
);

// Set cURL options
curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($data));
curl_setopt($ch, CURLOPT_HTTPHEADER, array(
    'content-type: application/json',
    'api-key: ' . $apiKey
));

$response = curl_exec($ch);

// Check for errors
if ($response === false) {
    echo 'cURL Error: ' . curl_error($ch);
} else {

    // echo $response; // Raw response
    // Decode the JSON response
    $responseData = json_decode($response, true);
    
    // Check if the response contains the expected data
    if (isset($responseData['choices']) && is_array($responseData['choices']) && !empty($responseData['choices'])) {
        $aiResponse = $responseData['choices'][0]['message']['content'];
        echo 'AI Response: ' . $aiResponse;
    } else {
        echo 'Unexpected response format or empty response.';
    }    
}


// Close the cURL session
curl_close($ch);
?>